x-pack/solutions/security/plugins/security_solution/public/timelines/store/middlewares/timeline_save.ts

```diff
@@ -51,124 +53,159 @@ function isSaveTimelineAction(action: Action): action is ReturnType<typeof saveT
-        storeState,
-        SourcererScopeName.timeline
-      );
+    const { id: localTimelineId } = action.payload;
+    const storeState = store.getState();
+    const timeline = selectTimelineById(storeState, localTimelineId);
+    const { timelineId, timelineVersion, templateTimelineId, templateTimelineVersion } =
+      extractTimelineIdsAndVersions(timeline);
+    const timelineTimeRange = inputsSelectors.timelineTimeRangeSelector(storeState);
+    const selectedDataViewIdSourcerer = sourcererSelectors.sourcererScopeSelectedDataViewId(
+      storeState,
+      SourcererScopeName.timeline
+    );
+    const selectedPatternsSourcerer = sourcererSelectors.sourcererScopeSelectedPatterns(
+      storeState,
+      SourcererScopeName.timeline
+    );
 
-      store.dispatch(startTimelineSaving({ id: localTimelineId }));
+    const { dataViewId: experimentalDataViewId } = sourcererAdapterSelector(
+      SourcererScopeName.timeline
+    )(storeState);
 
-      try {
-        const response = await (action.payload.saveAsNew && timeline.id
-          ? copyTimeline({
-              timelineId,
-              timeline: {
-                ...convertTimelineAsInput(timeline, timelineTimeRange),
-                dataViewId: selectedDataViewIdSourcerer,
-                indexNames: selectedPatterns,
-                templateTimelineId,
-                templateTimelineVersion,
-              },
-              savedSearch: timeline.savedSearch,
-            })
-          : persistTimeline({
-              timelineId,
-              version: timelineVersion,
-              timeline: {
-                ...convertTimelineAsInput(timeline, timelineTimeRange),
-                dataViewId: selectedDataViewIdSourcerer,
-                indexNames: selectedPatterns,
-                templateTimelineId,
-                templateTimelineVersion,
-              },
-              savedSearch: timeline.savedSearch,
-            }));
+    const experimentalIsDataViewEnabled =
+      storeState.app.enableExperimental.newDataViewPickerEnabled;
 
-        if (isTimelineErrorResponse(response)) {
-          const error = getErrorFromResponse(response);
-          switch (error?.errorCode) {
-            case 403:
-              store.dispatch(showCallOutUnauthorizedMsg());
-              break;
-            // conflict
-            case 409:
-              kibana.notifications.toasts.addDanger({
-                title: i18n.TIMELINE_VERSION_CONFLICT_TITLE,
-                text: i18n.TIMELINE_VERSION_CONFLICT_DESCRIPTION,
-              });
-              break;
-            default:
-              kibana.notifications.toasts.addDanger({
-                title: i18n.UPDATE_TIMELINE_ERROR_TITLE,
-                text: error?.message ?? i18n.UPDATE_TIMELINE_ERROR_TEXT,
-              });
-          }
-          return;
-        }
+    let experimentalSelectedPatterns: string[] = [];
+
+    // NOTE: remove eslint override above after the experimental picker is stabilized
```

Comment:
Random, but do you have a label or something for tracking all of these `notes`? I.e. `NEW_DV_PICKER: NOTE: ......` so we can just search for all instances of the NEW_DV_PICKER later on to make sure all of these are accounted for?

Reply:
okay!